src/engines/TimeToLive.js

```diff
@@ -0,0 +1,91 @@
+import DefaultHashTable from '../dataStructure/HashTable';
+import DoublyLinkedList from '../dataStructure/DoublyLinkedList';
+import * as hashTableProp from '../hashTableSymbol';
+
+function TimeToLive({ HashTable = DefaultHashTable, defaultTTL } = {}) {
+  const store = new HashTable();
+  const timeSeriesIndex = new HashTable();
+  const timeIndexInterval = 5 * 60 * 1000; // milliseconds.
+
+  let lastRunGC = Date.now();
+
+  this.add = (key, value, ttl = defaultTTL) => {
+    if (!ttl)
+      throw Error(
+        'Expected ttl value. you can have to mention it in add method or mention as defaultTTL at constructor',
+      );
+
+    const expireTTL = Date.now() + ttl;
+    const bucket = getTimeBucket(expireTTL);
+    bucket.addFirst(key);
+    const tNode = bucket.getFirst();
```

Comment:
Do I miss something or is key === tNode

Reply:
Renamed the `getFirst` method to `getFirstNode`